### PR TITLE
Feature/data deletion update expiration

### PR DESCRIPTION
**Pull request by sankalpatimilsina12**

### CODE REVIEW

**Comment by sankalpatimilsina12** [comment]
This commit contains updates for handling data deletion, and file expiration. The commit doesn't contain NDN_DPDK related updates. That will be pushed to a separate branch. @susmit85

File: ndn_hydra/repo/utils/garbage_collector.py [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
     for file_name in files_to_remove:
-        # Delete from global view
+        # Delete from global view if not already deleted
         global_view.delete_file(file_name)
         logger.info(f"GARBAGE COLLECTOR: Removed {file_name} from global view.")
 
-        # TODO: Remove from data storage
+        # Remove from data storage and NDN-DPDK fileserver
+        data_storage.remove_packet(file_name) # TODO: need to remove all packets of the file
+        if os.path.exists(f"{config['fileserver_path']}/{file_name}"):
```
Comment (tntech-ngin) [inline ndn_hydra/repo/utils/garbage_collector.py]: Does this still need to be here? Basic Hydra has no notion of files. Only DPDK branch will have that.

Comment (sankalpatimilsina12) [reply ndn_hydra/repo/utils/garbage_collector.py]: @tntech-ngin That is the outdated commit, the latest commit on the PR has separated NDN_DPDK. Also, the `remove_packet` is removing data from database, not files and the function was there from the beginning.

Comment (tntech-ngin) [reply ndn_hydra/repo/utils/garbage_collector.py]: got it, thanks!

**Pull request merged**